### ISSUE TITLE
last() should reject the promise on a stream error?

### ISSUE THREAD

**Issue by djanowski** [lewisdiamond/stromjs#4]
Consider this example:

```js
import { Readable } from 'node:stream';
import strom from 'stromjs';

const stream = Readable.from([1]).pipe(
  strom.map(async item => {
    throw new Error('Foo');
  })
);

strom
  .last(stream)
  .then(() => console.log('Should not have resolved promise'))
  .catch(() => console.log('Promise correctly rejected'));
```

Thoughts?

**Comment by lewisdiamond** [comment]
Should be fixed in 0.5.5. Let me know if that works for you.

By the way, are you using Stromjs with Typescript or plain Javascript?

**Comment by djanowski** [comment]
Thank you for addressing this so quickly!

On this project I'm using plain JS.
